### PR TITLE
(O3-3313) : Supports integration of stock operations by referencing stock items to an external source.

### PR DESCRIPTION
**Pull request by jabahum**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR adds stock item references to a stock item

## Screenshots

image

https://github.com/openmrs/openmrs-esm-stock-management/assets/24471994/ace7ed68-a3cd-4853-a45e-0ba9368f85c0

## Related Issue

https://openmrs.atlassian.net/browse/O3-3313

### CODE REVIEW

**Comment by makombe** [comment]
@jabahum please check the ticket, @ojwanganto has left e comment. Also, if you can share video to illustrate the changes you have done will appreciate. Thanks

**Comment by jabahum** [comment]
> @jabahum please check the ticket, @ojwanganto has left e comment. Also, if you can share video to illustrate the changes you have done will appreciate. Thanks

This is noted let me share one

File: src/stock-items/add-stock-item/stock-item-references/stock-item-references.component.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        const err = extractErrorMessagesFromResponse(error);
+        showNotification({
+          title: t(
+            "deleteStockItemReferenceTitle",
+            `Error Deleting a stockitem reference`
+          ),
+          kind: "error",
+          critical: true,
+          description: err.join(","),
+        });
```
Comment (makombe) [inline src/stock-items/add-stock-item/stock-item-references/stock-item-references.component.tsx:214]: Can we change `showNotifications and showToast` to use snackbar for uniformity purposes

**Review by makombe — APPROVED** [review]
Also, can you ensure you mutate adding the reference code operation after saving. I see from the video you are navigating away from the page to another tab in order to see your changes.

File: src/stock-items/add-stock-item/stock-item-references/stock-item-references.component.tsx (hunk 1 above)
Comment (jabahum) [reply src/stock-items/add-stock-item/stock-item-references/stock-item-references.component.tsx:214]: this is noted let me work on this

**Comment by jabahum** [comment]
> Also, can you ensure you mutate adding the reference code operation after saving. I see from the video you are navigating away from the page to another tab in order to see your changes.

noted

**Pull request merged**